src/components/application_manager/rpc_plugins/vehicle_info_plugin/test/vehicle_info_pending_resumption_test.cc

```diff
@@ -0,0 +1,604 @@
+    MockApplicationManager;
+typedef NiceMock< ::test::components::application_manager_test::MockApplication>
+    MockApplication;
+typedef NiceMock<test::components::application_manager_test::MockRPCService>
+    MockRPCService;
+typedef std::shared_ptr<MockApplication> MockAppPtr;
+typedef std::shared_ptr<vehicle_info_plugin::VehicleInfoAppExtension>
+    VehicleInfoAppExtensionPtr;
+using hmi_apis::FunctionID::VehicleInfo_SubscribeVehicleData;
+
+class SubscribeCatcher {
+ public:
+  MOCK_METHOD2(subscribe,
+               void(const int32_t, const resumption::ResumptionRequest));
+};
+
+class VehicleInfoPendingResumptionHandlerTest : public ::testing::Test {
+  // using namespace vehicle_info_plugin;
+ public:
+  VehicleInfoPendingResumptionHandlerTest()
+      : mock_message_helper_(
+            *application_manager::MockMessageHelper::message_helper_mock())
+
+  {}
+
+  void SetUp() OVERRIDE {
+    ON_CALL(app_manager_mock_, event_dispatcher())
+        .WillByDefault(ReturnRef(event_dispatcher_mock_));
+    ON_CALL(app_manager_mock_, GetRPCService())
+        .WillByDefault(ReturnRef(mock_rpc_service_));
+    resumption_handler_.reset(
+        new vehicle_info_plugin::VehicleInfoPendingResumptionHandler(
+            app_manager_mock_));
+  }
+
+  MockAppPtr CreateApp(uint32_t app_id) {
+    auto mock_app = std::make_shared<MockApplication>();
+
+    ON_CALL(app_manager_mock_, application(app_id))
+        .WillByDefault(Return(mock_app));
+    ON_CALL(*mock_app, app_id()).WillByDefault(Return(app_id));
+    return mock_app;
+  }
+
+  VehicleInfoAppExtensionPtr CreateExtension(MockApplication& app) {
+    auto ext_ptr = std::make_shared<VehicleInfoAppExtension>(plugin_, app);
+    ON_CALL(app,
+            QueryInterface(VehicleInfoAppExtension::VehicleInfoAppExtensionUID))
+        .WillByDefault(Return(ext_ptr));
+    return ext_ptr;
+  }
+
+  resumption::Subscriber& get_subscriber() {
+    static resumption::Subscriber subscriber =
+        [this](const int32_t corr_id,
+               const resumption::ResumptionRequest resumption_request) {
+          this->subscribe_catcher_.subscribe(corr_id, resumption_request);
+        };
+    return subscriber;
+  }
+
+  SubscribeCatcher subscribe_catcher_;
+  MockMessageHelper& mock_message_helper_;
+  MockApplicationManager app_manager_mock_;
+  MockEventDispatcher event_dispatcher_mock_;
+  MockRPCService mock_rpc_service_;
+  vehicle_info_plugin::VehicleInfoPlugin plugin_;
+
+  std::unique_ptr<vehicle_info_plugin::VehicleInfoPendingResumptionHandler>
+      resumption_handler_;
+};
+
+smart_objects::SmartObjectSPtr CreateVDRequest(const uint32_t corr_id) {
+  using namespace application_manager;
+  smart_objects::SmartObjectSPtr request =
+      std::make_shared<smart_objects::SmartObject>(
+          smart_objects::SmartType_Map);
+  smart_objects::SmartObject& object = *request;
+  object[strings::params][strings::message_type] =
+      static_cast<int>(0);  // request
```

Comment:
Please replace magic numbers to valid constants or enums

Reply:
Replaced